lib/elixir/lib/task/supervisor.ex

```diff
@@ -67,6 +67,7 @@ defmodule Task.Supervisor do
   def async(supervisor, module, fun, args) do
     args = [self, :link, get_info(self), {module, fun, args}]
     {:ok, pid} = Supervisor.start_child(supervisor, args)
+    Process.link(pid)
```

Comment:
The following can happen:
* Supervisor starts task and sends task pid to caller
* Caller receives task pid and links to it (this can happen at any point from here)
* Supervisor receives :EXIT from parent and sends exit signal to task pid
* Task handles the exit signal and exits abruptly before it can link to caller

Because the start is now asynchronous, the easiest way for each process to ensure it is linked to the other is for both to link to the other at the first point it is required.

We have to link in the task too because the caller could be long dead before the task process is even spawned if the supervisor has a long message queue.